experimental/traffic-portal/src/app/core/cache-groups/asns/detail/asn-detail.component.ts

```diff
@@ -0,0 +1,110 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { Location } from "@angular/common";
+import { Component, OnInit } from "@angular/core";
+import { MatDialog } from "@angular/material/dialog";
+import { ActivatedRoute } from "@angular/router";
+import { ResponseASN, ResponseCacheGroup } from "trafficops-types";
+
+import { CacheGroupService } from "src/app/api";
+import { DecisionDialogComponent } from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import { NavigationService } from "src/app/shared/navigation/navigation.service";
+
+/**
+ * AsnDetailComponent is the controller for the ASN add/edit form.
+ */
+@Component({
+	selector: "tp-asn-detail",
+	styleUrls: ["./asn-detail.component.scss"],
+	templateUrl: "./asn-detail.component.html"
+})
+export class AsnDetailComponent implements OnInit {
+	public new = false;
+	public asn!: ResponseASN;
+	public cachegroups!: Array<ResponseCacheGroup>;
+	constructor(private readonly route: ActivatedRoute, private readonly cacheGroupService: CacheGroupService,
+		private readonly location: Location, private readonly dialog: MatDialog,
+		private readonly header: NavigationService) {
+	}
+
+	/**
+	 * Angular lifecycle hook where data is initialized.
+	 */
+	public async ngOnInit(): Promise<void> {
+		this.cachegroups = await this.cacheGroupService.getCacheGroups();
+		const ID = this.route.snapshot.paramMap.get("id");
+		if (ID === null) {
+			console.error("missing required route parameter 'id'");
+			return;
+		}
+
+		if (ID === "new") {
+			this.header.headerTitle.next("New ASN");
+			this.new = true;
+			this.asn = {
+				asn: 1,
+				cachegroup: "test",
+				cachegroupId: 1,
+				id: 1,
+				lastUpdated: new Date()
+			};
+			return;
+		}
+		const numID = parseInt(ID, 10);
+		if (Number.isNaN(numID)) {
+			console.error("route parameter 'id' was non-number:", ID);
+			return;
+		}
+
+		this.asn = await this.cacheGroupService.getASNs(numID);
+		this.header.headerTitle.next(`ASN: ${this.asn.asn}`);
+	}
+
+	/**
+	 * Deletes the current ASN.
+	 */
+	public async deleteAsn(): Promise<void> {
+		if (this.new) {
+			console.error("Unable to delete new ASN");
+			return;
+		}
+		const ref = this.dialog.open(DecisionDialogComponent, {
+			data: {message: `Are you sure you want to delete ASN ${this.asn.asn} with id ${this.asn.id}`,
+				title: "Confirm Delete"}
+		});
+		ref.afterClosed().subscribe(result => {
+			if(result) {
+				this.cacheGroupService.deleteASN(this.asn.id);
+				this.location.back();
+			}
+		});
+	}
+
+	/**
+	 * Submits new/updated ASN.
+	 *
+	 * @param e HTML form submission event.
+	 */
+	public async submit(e: Event): Promise<void> {
+		e.preventDefault();
+		e.stopPropagation();
+		if(this.new) {
+			this.asn = await this.cacheGroupService.createASN(this.asn);
+			this.new = false;
+		} else {
+			this.asn = await this.cacheGroupService.updateASN(this.asn);
```

Comment:
similarly, when this happens the title doesn't update if I change the ASN. So, for instance, if I change the ASN from 1 to 2, the ASN will have `asn: 2` but the title says "ASN: 1".

Reply:
I see the same happening for every other page, like division, region etc.
<img width="1126" alt="image" src="https://user-images.githubusercontent.com/22248619/226904454-5375b6ea-647f-4040-8f10-bc701fab70a9.png">
<img width="1142" alt="image" src="https://user-images.githubusercontent.com/22248619/226904602-d930e3ca-8025-44d1-afb3-f9817e9debe5.png">
One has to refresh the page to see the change.

Reply:
that's a problem. I guess there's no need for you to fix it since whoever does wind up fixing it will need to look over every such page anyway.

Reply:
Issue created: https://github.com/apache/trafficcontrol/issues/7418